src/maven/peakdetectiondialog.cpp

```diff
@@ -16,6 +16,12 @@ PeakDetectionDialog::PeakDetectionDialog(QWidget *parent) :
 
     setModal(true);
 
+    scoringSettingsDialog = new MSMSScoringSettingsDialog(this);
+    scoringSettingsDialog->setWindowFlags(scoringSettingsDialog->windowFlags() | Qt::WindowStaysOnTopHint);
+    connect(scoringSettingsDialog->btnOK, SIGNAL(clicked(bool)), scoringSettingsDialog, SLOT(hide()));
+    connect(btnFragMatchingAdvanced, SIGNAL(clicked()), scoringSettingsDialog, SLOT(bringIntoView()));
+    connect(scoringSettingsDialog->btnLoadClassAdduct, SIGNAL(clicked()), scoringSettingsDialog, SLOT(setLipidClassAdductFile()));
+
```

Comment:
Certain components publish signals, which are received by other components and processed in specific ways defined in "slots". Components listen for specific signals, and carry out the appropriate slot when they receive the signal. Very similar to pub/sub or observer pattern.